src/core.c/CompUnit/Repository/Installation.rakumod

```diff
@@ -744,6 +808,239 @@ Please ask the author to add a \"provides\" section, mapping every exposed names
     }
+
+    # Runner generation logic starts here.
+
+    my %arg-op = missing   => 0,
+                 plain     => 1,
+                 cmd-args  => 2,
+                 plat-sep  => 3,
+                 abs       => 4,
+                 abs-slash => 5,
+    ;
+
+    sub validate-env-var($name) {
+        die "$name is not a valid env var name" unless $name ~~ / ^ <[ a..z A..Z _ ]> <[ a..z A..Z 0..9 _ ]>* $ /;
+    }
+
+    sub generate-posix(
+        Str           :$program!,
+        Str           :$cwd,
+                      :@args,
+                      :%env-add,
+                      :@env-remove,
+    ) {
+        my @call-parts = "exec";
+
+        sub sh-escape(Str $text) {
+            S/ ( \$ | \` | \" | \\ | \n ) / \\$1 / given $text;
+        }
+
+        sub process-path(Str() $path) {
+            if $path ~~ /
+                    "<" ( "plain" || "cmd-args" || "plat-sep" || "abs" || "abs-slash" ) ">"
+                    (.*)
+                / {
+                my $op = $0.Str;
+                my $val = $1.Str;
+                given $op {
+                    when "plain" | "plat-sep" {
+                        return '"' ~ sh-escape($val) ~ '"';
+                    }
+                    when "cmd-args" {
+                        return '"$@"';
+                    }
+                    when "abs" | "abs-slash" {
+                        return '"${DIR}/' ~ sh-escape($val) ~ '"';
+                    }
+                }
+            }
+            else {
+                note "Don't understand this arg: $path";
+                exit 1;
+            }
+        }
+
+        # Write program
+        unless $program ~~ / ^ "<" ( "plain" || "plat-sep" || "abs" ) ">" / {
+            note "Program is invalid. Must start with \"<plain>\", \"<plat-sep>\", \"<abs>\".";
+            exit 1;
+        }
+        @call-parts.push: process-path($program);
+
+        # Write args
+        for @args -> $arg {
+            @call-parts.push: process-path($arg);
+        }
+
+        my @lines;
+
+        # Write CWD
+        if $cwd {
+            unless $cwd ~~ / ^ "<" ( "plain" || "abs" ) ">" / {
+                note "CWD is invalid. Must start with \"<plain>\" or \"<abs>\".";
+                exit 1;
+            }
+            @lines.push: 'cd ' ~ process-path($cwd);
+        }
+
+        # Write env vars
+        for %env-add.keys.sort -> $key {
+            validate-env-var $key;
+            my $val = %env-add{$key};
+            @lines.push: "export $key=$val"
+        }
+
+        @env-remove.map: -> $name {
+            validate-env-var $name;
+            @lines.push: "unset $name"
+        }
+
+        @lines.push: @call-parts.join(" ");
+
+        
+        my $text = get-resource('posix-runner-tmpl').slurp;
+        $text ~= @lines.join("\n");
+        $text
+    }
+
+    sub get-resource($name) {
+        nqp::gethllsym('default', 'SysConfig').rakudo-home.IO.add("resources").add($name)
```

Comment:
I wonder if there is a better name we could use than `resources`. That path location is already quite closer to the existing `resources` directories that repositories use while not being used for quite the same thing.

Reply:
Done. I went with `templates`.